lib/ios/AirGoogleMaps/AIRGoogleMapPolyline.m

```diff
@@ -49,6 +52,39 @@ -(void)setStrokeWidth:(double)strokeWidth
   _strokeWidth = strokeWidth;
   _polyline.strokeWidth = strokeWidth;
 }
+-(void)setLineDashPattern:(NSArray <NSNumber *> *)lineDashPattern {
+  _lineDashPattern = lineDashPattern;
+}
+-(void)updateLineDashPattern:(double)scale {
+  
+  // scale = 1.0 / mapView.projection.pointsForMeters(1, atCoordinate: mapView.camera.target)
+//  NSArray *lengths = @[@25, @20];
+  NSMutableArray *scalePatterns = [[NSMutableArray alloc] init];
+  for(int i=0; i< _lineDashPattern.count; i++){
+    [scalePatterns addObject:@(_lineDashPattern[i].intValue * scale)];
+  }
+  _scale = scale;
+  NSArray *styles = @[[GMSStrokeStyle solidColor: _strokeColor],
+                      [GMSStrokeStyle solidColor:[UIColor clearColor]]];
+  _polyline.spans = GMSStyleSpans(_polyline.path, styles, scalePatterns, kGMSLengthRhumb);
+}
+
+-(void)updateLineDash:(double)scale {
+
+  NSMutableArray *scalePatterns = [[NSMutableArray alloc] init];
+  for(int i=0; i< _lineDashPattern.count; i++){
+    [scalePatterns addObject:@(_lineDashPattern[i].intValue * scale)];
+  }
+  NSArray *styles = @[[GMSStrokeStyle solidColor: _strokeColor],
+                      [GMSStrokeStyle solidColor:[UIColor clearColor]]];
+    if (_polyline) {
+        @try {
```

Comment:
That's brutal

Reply:
I should check if it's null right? would not neeed the try catch? not sure

Reply:
Try catch in ObjC is very expensive operation, because of safeguard, switching context etc. Yes, you can add guard checks as I did. Have a look at `configureStyleSpansIfNeeded()` function that I added in this file. Try to take best of both for your implementation.